internal/ls/hover.go

```diff
@@ -31,6 +31,12 @@ func (l *LanguageService) ProvideHover(ctx context.Context, documentURI lsproto.
 	if quickInfo == "" {
 		return lsproto.HoverOrNull{}, nil
 	}
+
+	// Calculate the applicable range for the hover
+	rangeNode := getNodeForQuickInfo(node)
```

Comment:
This gets called twice - once from within `getQuickInfoAndDocumentation`.

In Strada, this is named as `nodeForQuickInfo`. We should just calculate it above the call to `getQuickInfoAndDocumentation` and get rid of the call within that function.